app/src/main/java/com/kickstarter/viewmodels/RewardsFragmentViewModel.kt

```diff
@@ -78,42 +76,79 @@ class RewardsFragmentViewModel {
 
             val project = this.projectDataInput
                     .map { it.project() }
-
-            val backedReward = project
-                    .map { it.backing()?.let { backing -> getReward(backing) } }
-                    .filter { ObjectUtils.isNotNull(it) }
-                    .map { requireNotNull(it) }
-                    .distinctUntilChanged()
-
+            
             project
                     .filter { it.isBacking }
                     .map { indexOfBackedReward(it) }
                     .distinctUntilChanged()
                     .compose(bindToLifecycle())
                     .subscribe(this.backedRewardPosition)
 
-            val pledgeDataAndReason = this.projectDataInput
-                    .compose<Pair<ProjectData, Reward>>(Transformers.takePairWhen(this.rewardClicked))
-                    .map { pledgeDataAndPledgeReason(it.first, it.second) }
-                    .distinctUntilChanged()
+            subscribeRewardClicked()
```

Comment:
Split out the reward clicked setup

Reply:
I'm gonna be hyper picky with this sorry 😞 , we can ha a chat if you want but, could we not split the init of the viewModel in functions? ... until now every viewModel has been following the same approach and just pure functions in the RXChains. Doing this we are breaking that consistency through the app.
To resume, could we not have the subscribeRewardClicked function and bring back to the constructor all the RXChains?